feat(cmdline): Add format for := lua expr

mikesmithgh opened the pull request:
Hey @folke there was a new commit on nightly https://github.com/neovim/neovim/commit/a92b38934a2d00c13ee4d1969d994da15e0857ab with a shorthand syntax `:=` for `:lua` cmdline. This feature is to match the pattern and format with the lua icon.

My autoformatter added the whitespace, let me know if you want me to remove it. 

> The `:= {expr}` syntax can be used to evaluate a lua expression, as
  a shorter form of `:lua ={expr}`. `:=` and `:[range]=` without argument
  are unchanged. However `:=#` and similar variants using |ex-flags|
  are no longer supported.

folke commented:
Thank you for this PR!

I've changed the implementation quite a bit so that multiple patterns are now allowed for a cmdline item.
Also added better support for `:lua=`

mikesmithgh commented:
> Thank you for this PR!
> 
> I've changed the implementation quite a bit so that multiple patterns are now allowed for a cmdline item. Also added better support for `:lua=`

awesome, 👍 just gave it test and looks good to me.
